From 09a7f0a9c360fedf33c5e5bbd1bc1b81d740cd0d Mon Sep 17 00:00:00 2001
Message-Id: <09a7f0a9c360fedf33c5e5bbd1bc1b81d740cd0d.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: Sivakumar Thulasimani <sivakumar.thulasimani@intel.com>
Date: Wed, 28 Jan 2015 22:51:48 +0530
Subject: [PATCH 58/95] MUST_REBASE [VPG]: adf/intel/vlv: Brightness fix for
 edp

This patch fixes the issue of unable to change brightness in edp + dp
scenario.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed.

Issue: GMINL-5485
Change-Id: Iaad6517b5101ef1d955feff71219287a34b86e7a
Signed-off-by: Thulasimani, Sivakumar <sivakumar.thulasimani@intel.com>
---
 .../video/adf/intel/core/common/dp/gen_dp_pipe.c   |   15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/drivers/video/adf/intel/core/common/dp/gen_dp_pipe.c b/drivers/video/adf/intel/core/common/dp/gen_dp_pipe.c
index abd49f9..e0ea69b 100644
--- a/drivers/video/adf/intel/core/common/dp/gen_dp_pipe.c
+++ b/drivers/video/adf/intel/core/common/dp/gen_dp_pipe.c
@@ -494,6 +494,8 @@ static struct intel_pipe_ops dp_base_ops = {
 	.dpst_irq_handler = dp_pipe_dpst_irq_handler,
 };
 
+static struct intel_pipe_ops edp_base_ops;
+
 u32 dp_pipe_init(struct dp_pipe *dp_pipe, struct device *dev,
 	struct intel_plane *primary_plane, u8 idx,
 	struct intel_pipeline *pipeline, enum intel_pipe_type type)
@@ -507,17 +509,16 @@ u32 dp_pipe_init(struct dp_pipe *dp_pipe, struct device *dev,
 	dp_pipe->dpms_state = DRM_MODE_DPMS_OFF;
 
 	/* encoder init  */
-	if (type == INTEL_PIPE_DP) {
-		dp_base_ops.set_brightness = NULL;
-		dp_base_ops.get_brightness = NULL;
+	if (type == INTEL_PIPE_DP)
 		err = intel_pipe_init(intel_pipe, dev, idx, true,
 			INTEL_PIPE_DP, primary_plane, &dp_base_ops, "dp_pipe");
-	} else {
-		dp_base_ops.set_brightness = dp_set_brightness;
-		dp_base_ops.get_brightness = dp_get_brightness;
+	else {
+		edp_base_ops = dp_base_ops;
+		edp_base_ops.set_brightness = dp_set_brightness;
+		edp_base_ops.get_brightness = dp_get_brightness;
 
 		err = intel_pipe_init(intel_pipe, dev, idx, true,
-			INTEL_PIPE_EDP, primary_plane, &dp_base_ops,
+			INTEL_PIPE_EDP, primary_plane, &edp_base_ops,
 			"edp_pipe");
 	}
 
-- 
1.7.9.5

